src/modules/http-exchange.ts

```diff
@@ -23,6 +23,10 @@ const createAbortController = () => {
   return new AbortController();
 };
 
+export const defaultFetchOptions: RequestInit = Object.freeze({
```

Comment:
This doesn't significantly change anything as we already have `fetchOptions`, so I think it'd not make sense to customise this per call of fetch as you can just override most stuff. So letting the `httpExchange` override using `fetchOptions` and then introducing `defaultFetchOptions` doesn't add much value.

Also I think it's not worth using `Object.freeze` here.

Anyway, can we introduce defaults for `headers` instead as that's what we wanted to go for originally :)

Reply:
@kitten So just do default headers.

As for the freeze, since it is being passed into a function on the outside a developer could do a non-immutable change (ie not using spread operator) and it would have side effects.  To prevent that someone from making that mistake, I froze it so they would get a runtime exception.